### ISSUE TITLE
whl and deb links

### ISSUE THREAD

**Issue by usernameisntemail** [P2Enjoy/kohya_ss-docker#9]
Most of the specific versions of the .whl and .deb files mentioned in the dot files and readme respectively don't seem to be on the websites mentioned in the readme.

So I'm just going to use the ones from pytorch.org, developer.download.nvidia.com and pythonhosted.org. Some versions I couldn't find at all, so am trying the closest ones available on pypi.org.

**Comment by usernameisntemail** [comment]
That appeared to install, and seems to be running. (But now there's a problem loading localhost so I can't check if it actually trains, but bored of it so give up.)

**Comment by kenneth104** [comment]
(english is not my first language)

I can found .tensorflow, .torch, .torchvision,
But not xformers, and i need it (only 8G vRAM)

about only listen localhost,u can easy use "socat" to other 0.0.0.0 ports
nomal use is change the .py or change running parameters

**Comment by kenneth104** [comment]
1. In the end I used this version and tried to package xformers by myself and failed

https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/linux/xformers-0.0.14.dev0-cp310-cp310-linux_x86_64.whl


2.  In the libs file, use xxxxx8_8.5.3-1+cuda11.8


3. use "docker-compose --profile kohya up --build" to build the container


Build it's OK，Follow-up will check how to run

**Comment by martinobettucci** [comment]
Hi!

@kenneth104 @usernameisntemail, you can indeed install the nearest versions available: the xformer specially make reference to the git hash for the bravest who wants to build it by themselves.

You need to run the command with the `docker compose --profile kohya up --build` which specify the profile.
The command is in the homepage of the project!

**Comment by martinobettucci** [comment]
For those who can not access once launched, please use `http://127.0.0.1:7680` (as in the project homepage).
DO NOT use `http://localhost:7680` as this is NOT equivalent to 127.0.0.1 inside docker therefor it won't work.

**Comment by usernameisntemail** [comment]
> about only listen localhost,u can easy use "socat" to other 0.0.0.0 ports nomal use is change the .py or change running parameters

I may try on a different IP4. I'm using wsl2, which isn't very good at localhost judgiing by people's comments.

**Comment by usernameisntemail** [comment]
> For those who can not access once launched, please use `http://127.0.0.1:7680` (as in the project homepage). DO NOT use `http://localhost:7680` as this is NOT equivalent to 127.0.0.1 inside docker therefor it won't work.

With mine (docker on ubuntu on wsl on windows10 )  127.0.0.1 gives a connection refused error and localhost gives a not found error.

I can ping and wget the web interface index html from another shell inside wsl, probably I could install a browser to the wsl ubuntu instance (haven't tried). 

Can't get to localhost from windows host's browser. Maybe a wsl2 / windows problem but learning how to troubleshoot that sounds boring, so probably try to do it a different way.

**Comment by kenneth104** [comment]
@martinobettucci 
I'm use "docker compose --profile kohya up --build",But the problem still persists

check the "docker-compose.yml" and the has been built image "kohya-docker-kohya"
It's loss some command  or file is stored in the wrong location when building?

Thank you very much for your project


[+] Building 3.4s (26/26) FINISHED                                                                                                                     
 => [internal] load .dockerignore                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                   0.0s 
 => [internal] load build definition from Dockerfile                                                                                              0.0s 
 => => transferring dockerfile: 32B                                                                                                               0.0s 
 => resolve image config for docker.io/docker/dockerfile:1                                                                                        2.5s 
 => CACHED docker-image://docker.io/docker/dockerfile:1@sha256:39b85bbfa7536a5feceb7372a0817649ecb2724562a38360f4d6a7782a409b14                   0.0s
 => [internal] load build definition from Dockerfile                                                                                              0.0s 
 => => transferring dockerfile: 32B                                                                                                               0.0s 
 => [internal] load .dockerignore                                                                                                                 0.0s 
 => [internal] load metadata for docker.io/nvidia/cuda:11.6.2-cudnn8-devel-ubuntu20.04                                                            0.8s 
 => [internal] load build context                                                                                                                 0.0s
 => => transferring context: 862B                                                                                                                 0.0s 
 => [ 1/17] FROM docker.io/nvidia/cuda:11.6.2-cudnn8-devel-ubuntu20.04@sha256:eb5b849e629ecdcc924d0e0eb2405377792d7ca24d3bd627cc5792e763bff977    0.0s 
 => CACHED [ 2/17] RUN <<EOF (# apt for general container dependencies...)                                                                        0.0s 
 => CACHED [ 3/17] RUN <<EOF (# python...)                                                                                                        0.0s 
 => CACHED [ 4/17] COPY ./data/libs/*.deb /docker/                                                                                                0.0s 
 => CACHED [ 5/17] RUN <<EOF (# cuda...)                                                                                                          0.0s 
 => CACHED [ 6/17] WORKDIR /koyah_ss                                                                                                              0.0s 
 => CACHED [ 7/17] RUN <<EOF (git clone https://github.com/P2Enjoy/kohya_ss.git /koyah_ss...)                                                     0.0s 
 => CACHED [ 8/17] COPY ./data/*.whl /docker/                                                                                                     0.0s 
 => CACHED [ 9/17] COPY ./scripts/install-container-dep.sh ./data/*-requirements.txt /docker/                                                     0.0s 
 => CACHED [10/17] RUN <<EOF (# tensorflow, torch, torchvision, torchaudio...)                                                                    0.0s 
 => CACHED [11/17] RUN <<EOF (# xformers...)                                                                                                      0.0s 
 => CACHED [12/17] RUN <<EOF (# deepspeed...)                                                                                                     0.0s 
 => CACHED [13/17] RUN <<EOF (# Build requirements...)                                                                                            0.0s 
 => CACHED [14/17] RUN <<EOF (# jax/tpu...)                                                                                                       0.0s 
 => CACHED [15/17] RUN <<EOF (# fix LDCONFIG...)                                                                                                  0.0s 
 => CACHED [16/17] COPY ./scripts /docker/                                                                                                        0.0s 
 => CACHED [17/17] RUN <<EOF (chmod +x /docker/{run,mount,debug}.sh...)                                                                           0.0s 
 => exporting to image                                                                                                                            0.0s 
 => => exporting layers                                                                                                                           0.0s
 => => writing image sha256:e890ace1af0e4010e9f158051b5ab0204e8789172b73631e96aab20b3137fe67                                                      0.0s 
 => => naming to docker.io/library/kohya-docker-kohya                                                                                             0.0s
[+] Running 0/0                                                                                                                                        
 ⠋ Container kohya-docker-kohya-1  Creating                                                                                                       0.0s
Error response from daemon: No command specified

**Comment by kenneth104** [comment]
Surprise, when I rebuilt today, the new docker image was automatically used and it worked successfully

When building, you need to manually pull the base docker image
(docker pull nvidia/cuda:11.6.2-cudnn8-devel-ubuntu20.04)
(it's time-sensitive)

Build it's OK，Follow-up will check how to run

**Comment by kenneth104** [comment]
find some possible bugs in Dockerfile,still doesn't work

"COPY ./scripts/ /docker/" > "COPY ./scripts/*.sh /docker/"
"EXPOSE 7860" > "EXPOSE 7680"

**Comment by martinobettucci** [comment]
Hello @kenneth104,

thank you for your feedback!

**Comment by martinobettucci** [comment]
> > For those who can not access once launched, please use `http://127.0.0.1:7680` (as in the project homepage). DO NOT use `http://localhost:7680` as this is NOT equivalent to 127.0.0.1 inside docker therefor it won't work.
> 
> With mine (docker on ubuntu on wsl on windows10 ) 127.0.0.1 gives a connection refused error and localhost gives a not found error.
> 
> I can ping and wget the web interface index html from another shell inside wsl, probably I could install a browser to the wsl ubuntu instance (haven't tried).
> 
> Can't get to localhost from windows host's browser. Maybe a wsl2 / windows problem but learning how to troubleshoot that sounds boring, so probably try to do it a different way.

I'm sorry but we don't have Windows at all nor we plan to have any; this repository if for pure linux only: I'm sorry to disappoint you :-(

**Comment by LEv145** [comment]
For windows users: https://github.com/serpotapov/Kohya_ss-GUI-LoRA-Portable
